crates/subspace-farmer/src/plot.rs

```diff
@@ -255,6 +259,25 @@ impl Plot {
         self.piece_count() == 0
     }
 
+    /// Returns range which contains all of the pieces
```

Comment:
We have `IndexHashToOffsetDB.max_distance_cache`, I'm wondering if we can combine it with this use case to avoid reaching database unnecessarily. It depends on how often this method is called, just feels like we already have the information necessary in memory.

Reply:
We

Reply:
We actually have `max_distance_cache` but there is no guarantee that we will have `PieceIndexHashes` from both top and bottom.

I think this is not an issue, as we won't call it often. I think once we would need it more frequently, we can optimize it and merge with `max_distance_cache`.

Reply:
I'm afraid we might need it quite often eventually, but it is an optimization we can do later (TODO wouldn't hurt).